### PR TITLE
Added chain extension for dapps-staking

### PR DESCRIPTION
**Pull request by PierreOssun**
- Added proprieties in chain extension -> so in polkadotjs-UI 1 UNIT is 18 decimals
- Added dapp-staking: current_era() + general_era_info() + dapps_staking - bond_and_stake()

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -7,23 +7,32 @@
 include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
```
Comment (Maar-io) [inline runtime/src/lib.rs:8]: change 
`spec_version: 100` -> 2

Comment (Maar-io) [reply runtime/src/lib.rs:8]: on line 105

File: node/Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,7 @@ name = "swanky-node"
 log = { version = "0.4.8" }
```
Comment (Maar-io) [inline node/Cargo.toml:20]: increase pallet version

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -370,6 +379,104 @@ impl pallet_dapps_staking::traits::IsContract for SmartContract {
...
+
+impl ChainExtension<Runtime> for LocalChainExtension {
+	fn call<E: Ext>(func_id: u32, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
+	where
+		<E::T as SysConfig>::AccountId: UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]>,
+	{
+		match func_id {
+			// 20 is dapps_staking prefix and 01 is function number
+			// dapps_staking - current_era()
+			2001 => {
```
Comment (Maar-io) [inline runtime/src/lib.rs:399]: let's start using pallet numbers. dapps-staking is 34. That would make this function to be 3401
https://github.com/AstarNetwork/Astar/blob/88017efb43f838f7f89970366d35c89ade734082/runtime/astar/src/lib.rs#L753

**Comment by Maar-io** [comment]
Let's move the chain extension implementation from the `runtime/src/lib.rs`.
There will be extensions coming and that file needs to remain clean.
Ideally I see that we have `chain-extension` folders in `astar-frame`, which would be linked to swanky and to `Astar` runtimes.
No need to do it in this PR, just to keep it in mind

File: runtime/src/lib.rs (hunk 3 above)
Comment (PierreOssun) [reply runtime/src/lib.rs:399]: Great idea

**Pull request merged**